dds/DCPS/transport/rtps_udp/RtpsUdpSendStrategy.cpp

```diff
@@ -567,8 +638,7 @@ RtpsUdpSendStrategy::pre_send_packet(const ACE_Message_Block* plain)
       break;
     }
 
-    if (!ok || (octetsToNextHeader == 0 && msgId != RTPS::PAD
-                && msgId != RTPS::INFO_TS)) {
+    if (!ok || (octetsToNextHeader == 0 && msgId != RTPS::PAD && msgId != RTPS::INFO_TS)) {
```

Comment:
This is the third loop I've seen for walking through RTPS Submessages in this PR alone. Does it make sense to try to make a generic function / object to capture this logic in a generic way?